sentry-android-core/src/main/java/io/sentry/android/core/SpanFrameMetricsCollector.java

```diff
@@ -135,11 +136,15 @@ private void captureFrameMetrics(@NotNull final ISpan span) {
         return;
       }
 
-      // ignore spans with no finish date
-      final @Nullable SentryDate spanFinishDate = span.getFinishDate();
+      // Ignore spans with no finish date, but SentryTracer is not finished when executing this
+      // callback, yet, so in that case we use the current timestamp.
+      final @Nullable SentryDate spanFinishDate =
+          span instanceof SentryTracer ? new SentryNanotimeDate() : span.getFinishDate();
```

Comment:
My only doubt is here. In case of idle transactions, the end date could be trimmed, causing the transaction to have more frames than reality (just until we move to span-only). But this actually matches what we already have in perfV1.

Reply:
Yeah at least for spans, the finish date is [always set](https://github.com/getsentry/sentry-java/blob/61981dc1039b766300d6e38e82ccacb439387cb7/sentry/src/main/java/io/sentry/Span.java#L200) before invoking the [span finish callback](https://github.com/getsentry/sentry-java/blob/61981dc1039b766300d6e38e82ccacb439387cb7/sentry/src/main/java/io/sentry/Span.java#L236), which itself [invokes this callback](https://github.com/getsentry/sentry-java/blob/305baf5bd419920f4b80ff6f0075c13dd42c2b94/sentry/src/main/java/io/sentry/SentryTracer.java#L440). Well spotted, thanks for fixing this! Yeah I guess we have no mechanism right now to deal with trimming.